teps/0093-add-sign-verify-subcommand-to-the-cli.md

```diff
@@ -0,0 +1,161 @@
+---
+status: proposed
+title: Add sign and verify subcommand to the CLI
+creation-date: '2022-10-05'
+last-updated: '2022-10-05'
+authors:
+- '@Yongxuanzhang'
+- '@nadgowdas'
+- '@lukehinds'
+collaborators: []
+---
+
+# TEP-0093: Add sign and verify subcommand to the CLI
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases](#use-cases)
+  - [Requirements](#requirements)
+- [Proposal](#proposal)
+- [Design Details](#design-details)
+- [Alternatives](#alternatives)
+- [Implementation Plan](#implementation-plan)
+  - [Test Plan](#test-plan)
+  - [Implementation Pull Requests](#implementation-pull-requests)
+- [References](#references)
+<!-- /toc -->
+
+## Summary
+
+This proposal aims at adding `sign` and `verify` to Tekton CLI to help Trusted Resources users to sign and verify Tekton Resources.
+
+
+## Motivation
+
+There is a growing open-source ecosystem around tekton [catalog](https://github.com/tektoncd/catalog) and [artifactHUB](https://artifacthub.io), where number of tasks/pipelines are made available ready-to-use in our CI/CD. These tasks provides commonly used functions like `git-clone`, `curl`, etc. And also includes vendor specific functions like `aws-cli`, `openshift-client`. Such a marketplace is essential in practicing DRY principle. But, as we are building such an open-distribution framework, we need to ensure task distribution is secure.
+
+Objective in this proposal is to faciliate an easy and handy approach for task/pipeline providers to sign their files and for consumers to verify them using native `tkn cli` commands.
+
+Also, this complements [TEP-0091](https://github.com/tektoncd/community/blob/main/teps/0091-trusted-resources.md). TEP-0091 proposes to apply verification at the controller and the signing is suggested to be implemented into Tekton CLI. This proposal suggests an CLI extension for developers to sign and verify tasks/pipelines YAML files.
+
+### Goals
+
+1. Add the ability to run `tkn {object} sign` and `tkn {object} verify` fom the CLI. `object` currently should support `task` and `pipeline`. `Sign` will take the private key and the resource yaml file, sign the resource object and attach signature to the annotation.
+2. Support both key files and KMS key.
+3. Support v1beta1 Task and Pipeline
```

Comment:
Depending on the timeline of the implementation it might make sense to support `v1` instead, or both. 
I'm not convinced that the version of the API should be part of the goals.

Reply:
Sure! I will update it to include v1!